### PR TITLE
Merge SLE 15 SP1

### PR DESCRIPTION
**Pull request by imobachgs**
Merge #492 into `master`.

### CODE REVIEW

File: test/addon_product_test.rb [hunk 1]
```diff
@@ -49,7 +61,8 @@
           ProductFactory.create_product("kind" => :product,
            "name" => "new_product", "version" => "1.0",
            "arch" => "x86_64", "source" => "1",
-           "product_package" => "new_product-release")
+           "product_package" => "new_product-release",
+           "deps" => [])
```
Comment (imobachgs) [inline test/addon_product_test.rb]: I wonder whether it is possible that "deps" is missing. In that case, `Y2Packager::Resolvable#method_missing` will raise an exception.

```
NoMethodError:
       undefined method `deps' for #<Y2Packager::Resolvable:0x000055b04eb50848>
```

File: src/lib/y2packager/repository.rb [hunk 2]
```diff
@@ -103,13 +106,16 @@ def find(repo_id)
       # @param autorefresh [Boolean]      Is auto-refresh enabled for this repository?
       # @param url         [URI::Generic] Repository URL
```
Comment (schubi2) [inline src/lib/y2packager/repository.rb]: Could you please add the product_dir description too ?

File: src/lib/y2packager/repository.rb [hunk 3]
```diff
@@ -120,12 +126,14 @@ def create(name:, url:, enabled: true, autorefresh: true)
     # @param enabled     [Boolean]      Is the repository enabled?
     # @param autorefresh [Boolean]      Is auto-refresh enabled for this repository?
     # @param url         [URI::Generic] Repository URL
```
Comment (schubi2) [inline src/lib/y2packager/repository.rb]: please update the docu too.

File: src/modules/AddOnProduct.rb [hunk 4]
```diff
@@ -2088,16 +2093,20 @@ def renamed_at?(renames, old_name, new_name)
...
       # Dependencies are not included in this call
       products = Y2Packager::Resolvable.find(kind: :product)
       products.each do |product|
-        renames = names_from_product_packages(product.product_package)
-          .reduce(renames) do |hash, rename|
-          add_rename_to_hash(hash, rename, product.name)
-        end
+        next unless product.product_package
+
+        @libzypp_product_renames = names_from_product_packages(product.product_package)
```
Comment (schubi2) [inline src/modules/AddOnProduct.rb]: Reduce is a nice stuff :-)

File: test/addon_product_test.rb (hunk 1 above)
Comment (schubi2) [reply test/addon_product_test.rb]: Yes, a good point.

**Comment by imobachgs** [comment]
@schubi2 Thanks for the review. I've updated the PR according to your suggestions. About the `#deps` thingie, I will open another PR against `yast2`.

**Pull request merged**